userapi/internal/api.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/devices"
+	"github.com/matrix-org/dendrite/userapi/api"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+type UserInternalAPI struct {
+	AccountDB  accounts.Database
+	DeviceDB   devices.Database
```

Comment:
Is the plan to do device management here too?

Reply:
Yes. The way these DBs are structured, accounts=profile info and devices=access tokens. Every public-facing component needs to check access tokens, and the majority of them also need access to profile info (client/fed api to service /profile requests, syncapi for access_token + profile data when `since` is empty, etc)